python/lsst/daf/ingest/ingestCatalogTask.py

```diff
@@ -0,0 +1,580 @@
+            return self.config.remap[fieldName]
+        return canonicalizeFieldName(fieldName)
+
+    def _columnDef(self, field):
+        """Return the SQL column definition for a given afw table field.
+        """
+        sqlType = fieldFormatters[field.getTypeString()].sqlType(field)
+        return self._columnName(field.getName()) + " " + sqlType
+
+    def _createTable(self, conn, tableName, schema):
+        """Create a table using column definitions based on the provided afw
+        table schema, adding in any extra columns specified in the config.  If
+        one exists, the unique id column is given a key.
+        """
+        fields = [item.field for item in self._schemaItems(schema)]
+        names = [f.getName() for f in fields]
+        equivalenceClasses = {}
+        for name in names:
+            equivalenceClasses.setdefault(name.lower(), []).append(name)
+        clashes = ',\n'.join('\t{' + ', '.join(c) + '}'
+                             for c in equivalenceClasses.itervalues() if len(c) > 1)
+        if clashes:
+            raise RuntimeError(
+                "Schema contains columns that differ only by non-word characters "
+                "and/or case:\n{}\nIn the database, these cannot be distinguished "
+                "and hence result in column name duplicates. Use the remap "
+                "configuration parameter to resolve this ambiguity.".format(clashes))
+        sql = "CREATE TABLE IF NOT EXISTS {} (\n\t".format(tableName)
+        sql += ",\n\t".join(self._columnDef(field) for field in fields)
+        if self.config.extraColumns:
+            sql += ",\n\t" + self.config.extraColumns
+        if self.config.idFieldName:
+            if self.config.idFieldName in names:
+                sql += ",\n\tUNIQUE({})".format(self._columnName(self.config.idFieldName))
+            else:
+                self.log.warn(
+                    "No field matches the configured unique ID field name "
+                    "({})".format(self.config.idFieldName))
+        sql += "\n)"
+        self._executeSql(conn, sql)
+
+    def _createView(self, conn, tableName, viewName, schema):
+        """Create a view that allows table columns to be referred to by their
+        afw table field aliases.
+        """
+        sql = ("CREATE OR REPLACE "
+               "ALGORITHM = MERGE SQL SECURITY INVOKER "
+               "VIEW {} AS SELECT\n\t").format(viewName)
+        with closing(conn.cursor()) as cursor:
+            cursor.execute("SHOW COLUMNS FROM " + tableName)
+            columnNames = [row[0] for row in cursor.fetchall()]
+        sql += ",\n\t".join(columnNames)
+        # Technically, this isn't quite right. In afw, it appears to be legal
+        # for an alias to shadow an actual field name. So for full rigor,
+        # shadowed field names would have to be removed from the columnNames
+        # list.
+        #
+        # For now, construct an invalid view and fail in this case.
```

Comment:
I think it's fine to leave this for now, but would it be worth filing it as an issue that ought to be addressed in future?